core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -2274,15 +2246,27 @@ public function editAll($intId=null, $ajaxId=null)
 		{
 			$class = 'tl_tbox';
 
+			if (Input::post('FORM_SUBMIT') == $this->strTable)
+			{
+				$this->Database->beginTransaction();
+			}
```

Comment:
I think we should wrap the code until line 2414 https://github.com/contao/contao/blob/16c02503f7e143257f7c1f380471b039b4256852/core-bundle/src/Resources/contao/drivers/DC_Table.php#L2413-L2415 with a try catch block, call `rollbackTransaction()` in the `catch(\Throwable)` and rethrow the exception.